connect/runtime/src/main/java/org/apache/kafka/connect/runtime/WorkerConnector.java

```diff
@@ -53,36 +56,67 @@
 
     private Map<String, String> config;
     private State state;
+    private OffsetStorageReader offsetStorageReader;
```

Comment:
How about `final` for the new field, and for any other fields that are set in the constructor?